File: Project/OPUS_CONTAINER_4_0_IZA/WEB_MAIN/js/common/ConstantMgr.js
var ___ConstantMgr = function() {};

var CONSTANT_MGR_isHeadOfficeCode = function(param ){
	var code = CONSTANT_MGR_getBaseLocationCode();
	if( code == param ){
		return true;
	}
	return false;
}

var CONSTANT_MGR_isBaseLocationCode = function(param ){
	var code = CONSTANT_MGR_getHeadOfficeCode();
	if( code == param ){
		return true;
	}
	return false;
}


var CONSTANT_MGR_isCompanyCode = function(param ){
	var code = CONSTANT_MGR_getCompanyCode();
	if( code == param ){
		return true;
	}
	return false;
}

var CONSTANT_MGR_isCompanyName = function(param ){
	var code = CONSTANT_MGR_getCompanyName();
	if( code == param ){
		return true;
	}
	return false;
}


var CONSTANT_MGR_isScacCode = function(param ){
	var code = CONSTANT_MGR_getScacCode();
	if( code == param ){
		return true;
	}
	return false;
}


var CONSTANT_MGR_getBaseLocationCode = function(){
	//var OPCC00001 = "ABC" ;
	var tmpCode = "";
	try{
		eval( " var tmpCode = OPC001 ;");
	}catch(e){
		alert("<%=ConstantMgr.getBaseLocationCodeToJS()%> does not defined");
		return "";
	}	
	return tmpCode;
}

var CONSTANT_MGR_getHeadOfficeCode = function(){
	//var OPCC00002 = "ABC" ;
	var tmpCode = "";
	try{
		eval( " var tmpCode = OPC002 ;");
	}catch(e){
		alert("<%=ConstantMgr.getHeadOfficeCodeToJS()%> does not defined");
		return "";
	}	
	return tmpCode;
}

var CONSTANT_MGR_getCompanyCode = function(){
	//var OPCC00003 = "ABC" ;
	var tmpCode = "";
	try{
		eval( " var tmpCode = OPC003 ;");
	}catch(e){
		alert("<%=ConstantMgr.getCompanyCodeToJS()%> does not defined");
		return "";
	}	
	return tmpCode;
}

var CONSTANT_MGR_getCompanyName = function(){
	//var OPCC00003 = "ABC" ;
	var tmpCode = "";
	try{
		eval( " var tmpCode = OPC005 ;");
	}catch(e){
		alert("<%=ConstantMgr.getCompanyNameToJS()%> does not defined");
		return "";
	}	
	return tmpCode;
}


var CONSTANT_MGR_getScacCode = function(){
	//var OPCC00003 = "ABC" ;
	var tmpCode = "";
	try{
		eval( " var tmpCode = OPC004 ;");
	}catch(e){
		alert("<%=ConstantMgr.getScacCodeToJS()%> does not defined");
		return "";
	}	
	return tmpCode;
}
___ConstantMgr.prototype.isHeadOfficeCode = CONSTANT_MGR_isHeadOfficeCode;
___ConstantMgr.prototype.isBaseLocationCode = CONSTANT_MGR_isBaseLocationCode;
___ConstantMgr.prototype.isCompanyCode = CONSTANT_MGR_isCompanyCode;
___ConstantMgr.prototype.isCompanyName = CONSTANT_MGR_isCompanyName;
___ConstantMgr.prototype.isScacCode = CONSTANT_MGR_isScacCode;

___ConstantMgr.prototype.getBaseLocationCode = CONSTANT_MGR_getBaseLocationCode;
___ConstantMgr.prototype.getHeadOfficeCode = CONSTANT_MGR_getHeadOfficeCode;
___ConstantMgr.prototype.getCompanyCode = CONSTANT_MGR_getCompanyCode;
___ConstantMgr.prototype.getCompanyName = CONSTANT_MGR_getCompanyName;
___ConstantMgr.prototype.getScacCode = CONSTANT_MGR_getScacCode;

var ConstantMgr = new ___ConstantMgr();

File: Project/OPUS_CONTAINER_4_0_IZA/APP-INF/src/com/clt/apps/opus/bcm/sup/valuemanage/util/ConstantMgr.java
/*=========================================================
*Copyright(c) 2009 CyberLogitec
*@FileName : ConstantMgr.java
*@FileTitle : OPUS Constants Manager
*Open Issues :
*Change history :
*@LastModifyDate : 2011.02.10
*@LastModifier : 송민석
*@LastVersion : 1.0
* 2011.02.10 송민석
* 1.0 Creation
=========================================================*/
package com.clt.apps.opus.bcm.sup.valuemanage.util;

import java.util.HashMap;
import java.util.List;
import java.util.Map;

import com.clt.apps.opus.bcm.sup.valuemanage.basic.ValueManageBC;
import com.clt.apps.opus.bcm.sup.valuemanage.basic.ValueManageBCImpl;
import com.clt.apps.opus.bcm.sup.valuemanage.vo.SetupConstantListVO;
import com.clt.framework.core.layer.event.EventException;

/**
 * 상수 정보를 caching하고 그 정보를 관리한다. <br>
 *
 * @author Min Seok Song
 * @see ValueManageBC 
 * @since J2EE 1.6
 */
public class ConstantMgr {
	private static Map<String,String> constantMap = null;
	public final static String OPC_BASE_LOCATOIN = "OPC001";
	public final static String OPC_HEAD_OFFICE_CODE = "OPC002";
	public final static String OPC_COMPANY_CODE = "OPC003";
	public final static String OPC_SCAC_CODE = "OPC004";
	public final static String OPC_COMPANY_NAME = "OPC005";
	

	/**
	 * ConstantMgr 객체 생성<br>
	 * ConstantMgr의 instance 생성 방지하고 첫 loading시 DB정보를 cache함.<br>
	 */
	private ConstantMgr(){

	}
	
	static{
		constantMap = new HashMap<String,String>();
		try{
			ValueManageBC valueManageBC = new ValueManageBCImpl();
			List<SetupConstantListVO> list = valueManageBC.searchSetupConstantCodeList();
			int size = list.size();
			for(int i = 0 ; i < size ; i++ ){
				constantMap.put(list.get(i).getConsCd(), list.get(i).getConsValCtnt());
			}
		}catch(EventException e){
			
		}
		
	}
	private static String makeJSString(String key,String value){
		String opcValue = "var "+key+" = \""+value+"\";";
		return opcValue;
	}

	/**
	 * Local 시간을 계산하기 위해 기준시간 Location을 return한다.(OPC001) <br>
	 * EX ) KRSEL
	 * 
	 * @return String
	 * 
	 */	
	public static String getBaseLocationCode(){
		return constantMap.get(OPC_BASE_LOCATOIN);
	}
	
	/**
	 * Local 시간을 계산하기 위해 기준시간 Location을 Javascript 형태로  return한다. <br>
	 * EX) var OPC001 = "KRSEL";
	 * 
	 * @return String
	 * 
	 */	
	public static String getBaseLocationCodeToJS(){
		String v = getBaseLocationCode();
		String key = OPC_BASE_LOCATOIN;
		return makeJSString( key, v );
	}
	
	/**  
	 * tCode와 기준 Location이 일치 하는지 check한다. 
	 * 
	 * @param String tCode 비교 대상  Location Code
	 * @return boolean 일치 : true, 불일치 : false
	 */		
	public static boolean isBaseLocationCode(String tCode){
		boolean result = false;
		String opcValue = getBaseLocationCode();
		if( opcValue != null && opcValue.equals(tCode)){
			result = true;
		}
		return result;
	}	
	
	/**
	 * 본사 코드를 return한다.(OPC002) <br>
	 * EX) SELHO
	 * 
	 * @return String
	 * 
	 */	
	public static String getHeadOfficeCode(){
		return constantMap.get(OPC_HEAD_OFFICE_CODE);
	}
	
	
	/**
	 * 본사 코드를 Javascript 형태로 return한다. <br>
	 * EX) var OPC002 = "SELHO";
	 * 
	 * @return String
	 * 
	 */	
	public static String getHeadOfficeCodeToJS(){
		String v = getHeadOfficeCode();
		String key = OPC_HEAD_OFFICE_CODE;
		
		return makeJSString( key, v );
	}
		
	/**
	 * tCode와 본사 office code가 일치 하는지 check한다. 
	 * 
	 * @param String tCode 비교 대상  본사 office code
	 * @return boolean 일치 : true, 불일치 : false
	 */			
	public static boolean isHeadOfficeCode(String tCode){
		boolean result = false;
		String opcValue = getHeadOfficeCode();
		if( opcValue != null && opcValue.equals(tCode)){
			result = true;
		}		
		
		return result;
	}	
	
	/**
	 * 회사 Code(Carrier Code)를 return 한다.(OPC003 <br>
	 * EX) CLT
	 *  
	 * @return String
	 * 
	 */		
	public static String getCompanyCode(){
		return constantMap.get(OPC_COMPANY_CODE);
	}
	
	/**
	 * 회사 Code(Carrier Code)를 Javascript 형태로 return 한다. <br>
	 * EX) var OPC003 = "CLT";
	 * 
	 * @return String
	 * 
	 */		
	public static String getCompanyCodeToJS(){
		String v = getCompanyCode();
		String key = OPC_COMPANY_CODE;
		
		return makeJSString( key, v );		
	}	
	
	/**
	 * tCode와 회사 코드가 일치 하는지 check한다. 
	 * 
	 * @param String tCode 비교 대상 회사 코드가
	 * @return boolean 일치 : true, 불일치 : false
	 */			
	public static boolean isCompanyCode(String tCode){
		boolean result = false;
		String opcValue = getCompanyCode();
		if( opcValue != null && opcValue.equals(tCode)){
			result = true;
		}		
		
		return result;
	}			
	
	
	
	
	/**
	 * SCAC - SCAC - Standard Carrier Alpha Code를 return 한다.(OPC004) <br>
	 * EX) CLTC
	 *  
	 * @return String
	 * 
	 */		
	public static String getScacCode(){
		return constantMap.get(OPC_SCAC_CODE);
	}
	
	/**
	 * SCAC - SCAC - Standard Carrier Alpha Code를  Javascript 형태로 return 한다. <br>
	 * EX) var OPC004 = "CLTC";
	 * 
	 * @return String
	 * 
	 */		
	public static String getScacCodeToJS(){
		String v = getScacCode();
		String key = OPC_SCAC_CODE;
		
		return makeJSString( key, v );		
	}	
	
	/**
	 * tCode와 SCAC - SCAC - Standard Carrier Alpha Code가 일치 하는지 check한다. 
	 * 
	 * @param String tCode 비교 대상 회사 코드가
	 * @return boolean 일치 : true, 불일치 : false
	 */			
	public static boolean isScacCode(String tCode){
		boolean result = false;
		String opcValue = getScacCode();
		if( opcValue != null && opcValue.equals(tCode)){
			result = true;
		}		
		
		return result;
	}			
	

	
	
	/**
	 * 회사 이름를 return 한다.(OPC005) <br>
	 * EX) Cyberlogitec Shipping
	 *  
	 * @return String
	 * 
	 */		
	public static String getCompanyName(){
		return constantMap.get(OPC_COMPANY_NAME);
	}
	
	/**
	 * 회사 이름을 Javascript 형태로 return 한다. <br>
	 * EX) var OPC005 = "Cyberlogitec Shipping";
	 * 
	 * @return String
	 * 
	 */		
	public static String getCompanyNameToJS(){
		String v = getCompanyName();
		String key = OPC_COMPANY_NAME;
		
		return makeJSString( key, v );		
	}	
	
	/**
	 * tCode와 회사 코드가 일치 하는지 check한다. 
	 * 
	 * @param String tCode 비교 대상 회사 이름
	 * @return boolean 일치 : true, 불일치 : false
	 */			
	public static boolean isCompanyName(String tCode){
		boolean result = false;
		String opcValue = getCompanyName();
		if( opcValue != null && opcValue.equals(tCode)){
			result = true;
		}		
		
		return result;
	}			
	
	
	
	/**
	 * OPUS Constant Code에 해당하는 OPUS Constant Value를 return 한다. <br>
	 * 
	 * @return String
	 * 
	 */	
	public static String getValue(String opcCode){
		return constantMap.get(opcCode);
	}
	
	/**
	 * OPUS Constant Code에 해당하는 OPUS Constant Value를 javascript 형태로 return 한다. <br>
	 * 
	 * @return String
	 * 
	 */	
	public static String getValueToJS(String opcCode){
		return  makeJSString( opcCode, constantMap.get(opcCode) );
	}	
 
}
